test/plugins/validation/2and3/items-required-for-array-objects.test.js

```diff
@@ -190,17 +191,293 @@ describe('validation plugin - semantic - items required for array objects - Open
+    const res = validate({ jsSpec: spec }, config);
+    expect(res.errors.length).toEqual(0);
+    expect(res.warnings.length).toEqual(1);
+    expect(res.warnings[0].path).toEqual(
+      'components.schemas.TestObject.required[0]'
+    );
+    expect(res.warnings[0].message).toEqual(
+      "Schema properties specified as 'required' should be defined"
+    );
+  });
+
+  it('should return a warning when a model with an anyOf does not define a required property in all of its children', () => {
+    const spec = {
+      components: {
+        schemas: {
+          RefProp: {
+            properties: {
+              OptionalProperty: {
+                type: `string`
+              }
+            }
+          },
+          TestObject: {
+            type: 'object',
+            required: ['ImportantProperty'],
+            anyOf: [
+              {
+                $ref: '#/components/schemas/RefProp'
+              },
+              {
+                properties: {
+                  ImportantProperty: {
+                    type: 'string'
+                  }
+                }
+              }
+            ]
+          }
+        }
+      }
+    };
+
+    const res = validate({ jsSpec: spec }, config);
+    expect(res.errors.length).toEqual(0);
+    expect(res.warnings.length).toEqual(1);
+    expect(res.warnings[0].path).toEqual(
+      'components.schemas.TestObject.required[0]'
+    );
+    expect(res.warnings[0].message).toEqual(
+      "Schema properties specified as 'required' should be defined"
     );
+  });
+
+  it('should not return a warning when a model with a nested anyOf defines a required property in all of its children', () => {
+    const spec = {
+      components: {
+        schemas: {
+          NestedRefProp: {
+            properties: {
+              ImportantProperty: {
+                type: `string`
+              }
+            }
+          },
+          RefProp: {
+            anyOf: [
+              {
+                $ref: '#/components/schemas/NestedRefProp'
+              },
+              {
+                properties: {
+                  ImportantProperty: {
+                    type: 'string'
+                  }
+                }
+              }
+            ]
+          },
+          TestObject: {
+            type: 'object',
+            required: ['ImportantProperty'],
+            anyOf: [
+              {
+                $ref: '#/components/schemas/RefProp'
+              },
+              {
+                properties: {
+                  ImportantProperty: {
+                    type: 'string'
+                  }
+                }
+              }
+            ]
+          }
+        }
+      }
+    };
+
+    const res = validate({ jsSpec: spec }, config);
+    expect(res.errors.length).toEqual(0);
+    expect(res.warnings.length).toEqual(0);
+  });
+
+  it('should not return a warning when a anyOf model with a nested allOf defines a required property in atleast one of its children', () => {
```

Comment:
```suggestion
  it('should not return a warning when a anyOf model with a nested allOf defines a required property in at least one of its children', () => {
```

Reply:
Good catch, I fixed this and must have accidentally reverted it haha